### PR TITLE
Feat/snapserver group UI

### PR DESCRIPTION
**Pull request by gsalinaslopez**
### Changes
- Sticky group headers for snapcast control list
- Snapcast control list composable and viewModel state list refactoring/simplification
- Separate `NowPlayingScreen` viewModel, being in charge of audio channel selection and snapcast control interaction
- Fix bug to remember last played snapserver on TuneIn Screen

### CODE REVIEW

**Review by guerman5 — APPROVED** [review]
This PR introduces the Snapserver group UI with sticky headers, volume controls, and better client grouping, making multi-device audio smoother. The ViewModel refactor and added tests look solid. With 35 files changed, we should keep an eye on edge cases.

File: app/src/main/java/tech/capullo/radio/ui/NowPlayingScreen.kt [hunk 1]
```diff
@@ -16,31 +16,49 @@ import androidx.compose.material3.Scaffold
...
-fun NowPlayingScreen(viewModel: TuneInViewModel = hiltViewModel()) {
-    val uiState by viewModel.tuneInState.collectAsStateWithLifecycle()
-    var showChannelDialog by remember { mutableStateOf(false) }
-    var selectedChannel by remember { mutableStateOf(AudioChannel.STEREO) }
+fun NowPlayingScreen(viewModel: NowPlayingViewModel = hiltViewModel()) {
+    val uiState by viewModel.nowPlayingUiState.collectAsStateWithLifecycle()
+    NowPlayingScreenContent(
+        uiState = uiState,
+        groups = viewModel.groups,
+        onClientVolumeChange = viewModel::onClientVolumeChange,
```
Comment (garpernica) [inline app/src/main/java/tech/capullo/radio/ui/NowPlayingScreen.kt:48]: SnarserverGroups call shows onClientVolumeChange twice in the diff. Once here and once in Line115. Make sure final code only passes this parameter once.

File: app/src/main/java/tech/capullo/radio/ui/NowPlayingScreen.kt [hunk 2]
```diff
@@ -91,39 +108,93 @@ fun NowPlayingScreen(viewModel: TuneInViewModel = hiltViewModel()) {
...
                     }
                 },
             )
         }
     }
 }
+
+@Preview(showBackground = true)
+@Composable
+fun NowPlayingScreenContentPreview() {
```
Comment (garpernica) [inline app/src/main/java/tech/capullo/radio/ui/NowPlayingScreen.kt:135]: The mock group/client construction is duplicated between Broadcaster and NowPlaying previews; you could extract a small helper to avoid repeating that structure, but that’s a minor nit.

**Review by garpernica — COMMENTED** [review]
Nice improvements, NowPlaying has its own ViewModel + UI state more testable and preview-friendly. Snapserver used from both Broadcaster and NowPlaying reduces duplication and keeps the Snapserver group UI consistent.

File: app/src/main/java/tech/capullo/radio/ui/NowPlayingScreen.kt (hunk 1 above)
Comment (gsalinaslopez) [reply app/src/main/java/tech/capullo/radio/ui/NowPlayingScreen.kt:48]: it may appear as being passed twice but in reality is being `hoisted` by the topmost `NowPlayingScreen` which in turns calls the function on the viewModel, making it the final point of action

File: app/src/main/java/tech/capullo/radio/ui/NowPlayingScreen.kt (hunk 2 above)
Comment (gsalinaslopez) [reply app/src/main/java/tech/capullo/radio/ui/NowPlayingScreen.kt:135]: Minor NIT agree.
